contracts/vaults/balancer/internal/pool/Boosted3TokenPoolUtils.sol

```diff
@@ -295,6 +295,7 @@ library Boosted3TokenPoolUtils {
 
         strategyTokensMinted = strategyContext._convertBPTClaimToStrategyTokens(bptMinted);
 
+        strategyContext.vaultState.totalBPTHeld += bptMinted;
```

Comment:
@weitianjie2000 just confirming, these changes will cover reward reinvestment and settlement, correct?

Reply:
I am updating it in both deposit and reward reinvestment. Settlement should be covered by redeem.